next-frontend/src/components/persons/RecordsTable.tsx

```diff
@@ -0,0 +1,66 @@
+import type { components } from "@/types/openapi";
+import { TFunction } from "i18next";
+import { Heading, VStack } from "@chakra-ui/react";
+import _ from "lodash";
+import { ByCompetitionTable } from "@/components/results/ResultsTable";
+import events from "@/lib/wca/data/events";
+
+export default function RecordsTable({
+  recordResults,
+  t,
+}: {
+  recordResults: components["schemas"]["Results"];
+  t: TFunction;
+}) {
+  const recordsByType = _.groupBy(
+    recordResults,
+    (r) => r.regional_single_record || r.regional_average_record,
+  );
+
+  return (
+    <VStack align="left">
+      {"WR" in recordsByType && (
+        <>
+          <Heading textStyle="h2">{t("persons.show.world_records")}</Heading>
+          <RecordsByEvent recordResults={recordsByType["WR"]} t={t} />
+        </>
+      )}
+      {["ER", "NAR", "SAR", "ASR", "OCR"].map((region) => {
+        return (
+          region in recordsByType && (
+            <>
```

Comment:
Needs to be a `React.Fragment` so that it can specify `key={region}`

Reply:
Yes and no, from a React perspective it does, but a user will never have more than one kind of Continental Record at a time. I am still adding it though.